scripts/check-code.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+# check-code.sh
+#
+# SUMMARY
+#
+#   Checks all Vector code
+
+set -exuo pipefail
+
+cargo check --all --all-targets
```

Comment:
on windows we have different feature flags we need to check so on windows this needs to:

`cargo check --no-default-features --features default-msvc`

There are also two other env flags that need to be set

https://vector.dev/docs/setup/installation/manual/from-source/#add-perl-to-your-path